### PR TITLE
Indicate Committable On SalesInvoice Call

### PR DESCRIPTION
**Pull request by HoyaBoya**
Need to indicate that the SalesInvoice is commit true on the end of a Spree checkout.

Related Asana
https://app.asana.com/0/12264068303259/23903008716699

### CODE REVIEW

File: spec/models/spree_avatax/invoice_spec.rb [hunk 1]
```diff
@@ -3,13 +3,34 @@
...
+  let(:doc_type) { SpreeAvatax::Invoice::SALES_ORDER }
   let(:params) { {doc_type: doc_type} }
   let(:order) { create(:order_with_line_items) }
   let(:tax_rate) { create(:tax_rate, calculator: calculator, zone: ZoneSupport.global_zone) }
   let(:invoice_instance) { SpreeAvatax::Invoice.new(order, doc_type) }
   let(:tax_category) { create(:tax_category) }
 
+  describe '.committable?' do
+    before do
+      allow_any_instance_of(SpreeAvatax::Invoice).to receive(:build_invoice)
```
Comment (evanwhalen) [inline spec/models/spree_avatax/invoice_spec.rb]: Why does this need to be stubbed?

Comment (HoyaBoya) [inline spec/models/spree_avatax/invoice_spec.rb]: ```build_invoice``` does a ton of work with the Spree object graph, and this is called in the object constructor for Invoice. I didn't think it was necessary to do all the setup work in the tests just to be able to invoke ```committable?```, hence the stub and I believe ```.new``` was tested in another suite.

Without the stub, there are tons of nil errors as build invoice tries building a tax request with a order and all of it's trimmings.

Comment (evanwhalen) [inline spec/models/spree_avatax/invoice_spec.rb]: Gotcha, I missed the build_invoice in the constructor, this makes sense.

**Comment by evanwhalen** [comment]
:+1:

File: spec/models/spree_avatax/invoice_spec.rb (hunk 1 above)
Comment (HoyaBoya) [inline spec/models/spree_avatax/invoice_spec.rb]: I did too :).

**Comment by HoyaBoya** [comment]
Merging.

**Pull request merged**